### PR TITLE
add hdf output to pycbc_page_foreground

### PR DESCRIPTION
**Pull request by GarethDaviesGW**

### CODE REVIEW

**Comment by GarethDaviesGW** [comment]
I've just realised you asked for this to be a method in the io.hdf module. I'll switch that around now

File: pycbc/io/hdf.py [hunk 1]
```diff
@@ -998,6 +997,56 @@ def to_coinc_xml_object(self, file_name):
...
+        # Single-detector fields
+        for field in ['chisq', 'chisq_dof', 'coa_phase', 'end_time',
+                      'sg_chisq', 'sigmasq', 'snr']:
+            vals_valid = self.get_snglfile_array_dict(field)
+            for ifo in self.ifos:
+                vals = vals_valid[ifo][0]
+                valid = vals_valid[ifo][1]
+                vals[np.logical_not(valid)] = -1.
+                ofd.create_dataset(ifo + '_'+field, data=vals,
+                                   dtype=np.float32)
```
Comment (spxiwh) [inline pycbc/io/hdf.py:1048]: Can you add another special column here. `network_snr`, which would be the quadrature sum of the various `snr` columns?

**Review by spxiwh — APPROVED** [review]
Thanks Gareth. I made two minor requests, once done I'm happy for this to be merged.

File: pycbc/io/hdf.py [hunk 2]
```diff
@@ -802,13 +804,10 @@ def get_snglfile_array_dict(self, variable):
...
-                times = self.get_coincfile_array('{}/time'.format(ifo))
-                if ref_times is None:
-                    ref_times = times
-                else:
-                    ref_times[ref_times < 0] = times[ref_times < 0]
+            times_gen = (self.get_coincfile_array('{}/time'.format(ifo))
+                         for ifo in ifos)
+            ref_times = np.array([mean_if_greater_than_zero(t)[0]
+                                  for t in zip(*times_gen)])
         except KeyError:  # Else fall back on old two-det format
```
[the remark this answers is not in the document]
Comment (GarethDaviesGW) [reply pycbc/io/hdf.py]: removed, (as well as one other place I happened to see this when searching the file for this spot)

File: pycbc/io/hdf.py [hunk 3]
```diff
@@ -998,6 +986,69 @@ def to_coinc_xml_object(self, file_name):
...
+
+        snr_vals_valid = self.get_snglfile_array_dict('snr')
+        network_snr_sq = np.zeros_like(snr_vals_valid[self.ifos[0]][0])
+        for ifo in self.ifos:
+            vals = snr_vals_valid[ifo][0]
+            valid = snr_vals_valid[ifo][1]
+            vals[np.logical_not(valid)] = -1.
+            ofd.create_dataset(ifo + '_snr', data=vals,
+                               dtype=np.float32)
+            network_snr_sq[valid] += vals[valid]
```
Comment (spxiwh) [inline pycbc/io/hdf.py:1047]: Don't you want a `(vals[valid])**2` here?

Comment (GarethDaviesGW) [reply pycbc/io/hdf.py]: thats what I just noticed and have put in

**Pull request merged**